Count % of reads mapped to reference
===

Once all the reads have been aligned to the reference genome using top_hat, use [percent_align.py](Scripts_HPC/percent_align.py) to obtain a csv file with the % of input reads aligned to reference genome (in this case, wheat)


```python percent_align.py (1) (2) (3)```

- (1) Directory where the reads are 

- (2) Name of text file with all the library names listed 

- (3) Name for the output csv file


```python percent_align.py 2015_isolates tail percent``



/tgac/workarea/collaborators/saunderslab/Wheat_IWGSC_transcripts