Reads mapping to wheat transcriptome
===

###2015 samples

Missing Filter/
```LIB18260
LIB18261
LIB18569
LIB18570
LIB18571
LIB18572
LIB18573
LIB18574
LIB18575
LIB18576
LIB18577
LIB18578
LIB18579
LIB18580```

###Wheat transcriptome alignment

To build the wheat transcriptome to map the RNA-seq reads against, I'm using the cdna.all from the Gene Annotation files in ENSEMBL. This fasta file includes the super-set of all transcripts resulting from Ensembl known, novel and pseudo gene predictions.

Run in /tgac/workarea/users/morenop

######Index reference genome 
 
```
bsub -q Test256 -We 10000 -o triticum_index_cdna.txt 'bowtie2-build Gene_annotation/cdna/Triticum_aestivum.IWGSC1.0+popseq.29.cdna.all.fa Triticum_aestivum_cdna.fa'
```

For tophat/bowtie to work, the names of the index files should be: reference.fa, reference.1.bt2, reference.2.bt2, etc. 

Skip the filter and the quality check as it was already done for the 2015 samples. 

```
bsub -q Test256 -We 10000  -o tophat.txt 'sh top_hat.sh'
```

###Tophat.sh

```
#!/bin/bash
source bowtie-2.2.5
source tophat-2.0.11
source samtools-0.1.19
while read line; do
    tophat -r 200 -p 2 -o 2015_isolates/$line/top_hat Ref_cdna_wheat/Triticum_aestivum_cdna 2015_isolates/$line/Filter/$line\_R1.fastq 2015_isolates/$line/Filter/$line\_R2.fastq
done < 2015_isolates/tmp.txt
```
```
while read line; do
	mkdir BAM_files 
    samtools sort $line/top_hat/accepted_hits.bam $line/BAM_files/accepted_hits_sorted
    samtools index $line/BAM_files/accepted_hits_sorted.bam
done < 2015_isolates/tmp.txt
``

Failed with the following error:

```
Mapping left_kept_reads to genome Triticum_aestivum_cdna with Bowtie2 
	[FAILED]
Error running bowtie:
Error while flushing and closing output
terminate called after throwing an instance of 'int'
(ERR): bowtie2-align died with signal 6 (ABRT) (core dumped)
```

Remove multi-threads (-p , -n) and change bowtie version:

```
source bowtie-2.2.1
source tophat-2.0.11
source samtools-0.1.19
bsub -q Test256 -We 10000  -o tophat.txt 'tophat -r 200 -o 2015_isolates/LIB18684/top_hat Ref_cdna_wheat/Triticum_aestivum_cdna 2015_isolates/LIB18684/Filter/LIB18684_R1.fastq 2015_isolates/LIB18684/Filter/LIB18684_R2.fastq'
```


###Top_Hat launcher

```
#!/bin/bash
 while read line; do
	echo "$line"
	IN1="2015_isolates/$line/Filter/${line}_R1.fastq"		#Right reads
	IN2="2015_isolates/$line/Filter/${line}_R2.fastq"		#Left reads
	OUT="2015_isolates/$line/top_hat"	
		#Output directory
	echo `bsub -q Test128 -J ${line}_tophat -o launcher.txt -We 100000 -R"rusage[mem=50000]" "source bowtie-2.2.1; source samtools-0.1.19; source tophat-2.0.11; tophat -r 200 -N 0 -o $OUT Ref_cdna_wheat/Triticum_aestivum_cdna $IN1 $IN2"`  
done < 2015_isolates/tmp.txt
```